src/Reader/Reader.php

```diff
@@ -189,17 +197,14 @@ public static function useHttpConditionalGet($bool = true)
      */
     public static function import($uri, $etag = null, $lastModified = null)
     {
-        $cache       = self::getCache();
-        $client      = self::getHttpClient();
-        $client->resetParameters();
-        $headers = new ZendHttp\Headers();
-        $client->setHeaders($headers);
-        $client->setUri($uri);
+        $cache   = self::getCache();
+        $client  = self::getHttpClient();
         $cacheId = 'Zend_Feed_Reader_' . md5($uri);
 
         if (static::$httpConditionalGet && $cache) {
-            $data = $cache->getItem($cacheId);
-            if ($data) {
+            $headers = [];
+            $data    = $cache->getItem($cacheId);
+            if ($data && $client instanceof Http\HeaderAwareClientInterface) {
```

Comment:
> But now non HeaderAwareClientInterface clients are not aware of this headers. $headers->addHeaderLine is never called for those clients.

I have no idea what you're getting at.

Yes, non-`HeaderAwareClientInterface` implementations won't get the headers. That's as intended; if the implementation is implementing the original `ClientInterface::get()`, then the second `$headers` argument we're passing *has no known meaning* for that implementation. As such, there's no reason to perform cache lookups, because the client is not known to be able to accept headers anyways.